src/UnitfulRecipes.jl

```diff
@@ -42,11 +47,15 @@ function resolve_axis!(attr, arr::AbstractArray{T}, axis::Int) where {T<:Quantit
 
     # get axis label and append unit
     key = key_label(axis)
+    ckey = (axis==3) ? :colorbar_title : key
     if haskey(attr, key)
         attr[key] = string(attr[key], " ($(string(u)))")
+        attr[ckey] = attr[key]
     else
         attr[key] = string(u)
+        attr[ckey] = attr[key]
```

Comment:
You do this on both branches of the if?

Reply:
Yes apologies this is a bit dirty... I just wanted a working prototype for `contourf` originally... I would need more time to make it neatly :/